PartyIcons/View/NameplateView.cs

```diff
@@ -106,18 +135,52 @@ public void SetupForPC(XivApi.SafeNamePlateObject npObject)
                             iconScale = 3f;
                             nameScale = 1.5f;
                             break;
+
+                        case NameplateSizeMode.Tiny:
+                            iconOffset = new Vector2(15, 74);
+                            iconScale = 1f;
+                            nameScale = 0.5f;
+                            break;
                     }
                     break;
 
                 case NameplateMode.RoleLetters:
-                    iconScale = 0f;
+                    if (_configuration.ShowPlayerStatus)
+                        switch (_configuration.SizeMode)
+                        {
+                            case NameplateSizeMode.Smaller:
+                            case NameplateSizeMode.Tiny:
+                                iconOffset = new Vector2(15, 74);
+                                iconScale = 1f;
+                                break;
+                            case NameplateSizeMode.Medium:
+                                iconOffset = new Vector2(0, 53);
+                                iconScale = 1.5f;
+                                break;
+                            case NameplateSizeMode.Bigger:
+                                iconOffset = new Vector2(-15, 34);
+                                iconScale = 2f;
+                                break;
+                        }
+                    else iconScale = 0f;
+
                     nameScale = _configuration.SizeMode switch
                     {
                         NameplateSizeMode.Smaller => 0.5f,
                         NameplateSizeMode.Medium  => 1f,
                         NameplateSizeMode.Bigger  => 1.5f,
+                        NameplateSizeMode.Tiny => 0.5f
                     };
                     break;
+            }
+
+            if (GetModeForNameplate(npObject) < NameplateMode.RoleLetters

```

Comment:
This condition is bloated and honestly feels like a quick fix. The changes should be better incorporated into the original code, not dumped on top of it.